bootstrap/terraform/main.tf

```
@@ -118,26 +85,45 @@ module "eks_blueprints" {
 #---------------------------------------------------------------
 
 module "eks_blueprints_kubernetes_addons" {
-  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons?ref=v4.18.1"
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons?ref=v4.22.0"
 
   eks_cluster_id = module.eks_blueprints.eks_cluster_id
 
   # Deploy Crossplane
+  # Default helm chart and providers values set at https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/modules/kubernetes-addons/crossplane/locals.tf
   enable_crossplane = true
 
-  crossplane_helm_config = local.crossplane_helm_config
+  #---------------------------------------------------------
+  # Crossplane community AWS Provider deployment
+  #---------------------------------------------------------
+  crossplane_aws_provider = {
+    enable = true
+    # to override the default irsa policy:
+    # additional_irsa_policies = ["arn:aws:iam::aws:policy/AmazonS3FullAccess"]
+  }
 
   #---------------------------------------------------------
-  # Crossplane AWS Provider deployment
-  #   Creates ProviderConfig name as "aws-provider-config"
+  # Crossplane Upbound AWS Provider deployment
   #---------------------------------------------------------
-  crossplane_aws_provider = local.crossplane_aws_provider
+  crossplane_upbound_aws_provider = {
+    enable = true
```

I think we need to set the provide config to `aws-provider-config`, if not the name will be `default`
```suggestion
    enable                = true
    provider_config = "aws-provider-config"
```

---

I think we should set this to "upbound-aws-provider-config".

---

Is should be `aws-provider-config` based on what's in the repo already using this value.

I think is better if we have both provider config named the same, in case we want to have a claim that can be use with aws provider or upbound provider based on label selectors. 

We are already have different files in the repo referring to this string value `aws-provider-config`
- https://github.com/awslabs/crossplane-on-eks/blob/main/bootstrap/eksctl/crossplane/upbound-aws-provider-config.yaml#L7
- https://github.com/awslabs/crossplane-on-eks/blob/main/examples/upbound-aws-provider/managed-resources/ec2/vpc.yaml#L14
- https://github.com/awslabs/crossplane-on-eks/blob/main/examples/upbound-aws-provider/managed-resources/s3/bucket.yaml#L15

Certainly something we can discuss on changing the name in all places in another PR after discussing some more